src/corehost/inc/pal.h

```diff
@@ -96,7 +98,7 @@ namespace pal
 
     bool realpath(string_t& path);
     bool file_exists(const string_t& path);
-    std::vector<pal::string_t> readdir(const string_t& path);
+    void readdir(const string_t& path, std::vector<pal::string_t>* list);
```

Comment:
Things like `pal::realpath` are using non-const references for in/out parameters, so it does work. Or are you just referring to a stylistic choice? As I said, I'm fine either way, just curious why pointers are preferred.